### ISSUE TITLE
no-multi-assign: misses assignment used as a (parenthesized) variable-declarator initializer (`const x = (obj.y = z)`)

### ISSUE THREAD

**Issue by ahmedelgabri** [oxc-project/oxc#25617]
### Summary

oxlint's `no-multi-assign` only fires when an `AssignmentExpression`'s right-hand side is itself an `AssignmentExpression` (classic `a = b = c = 1`). It does not fire when a `VariableDeclarator` initializer is a (parenthesized) `AssignmentExpression`, e.g. `const x = (obj.y = z)`. ESLint's own rule flags this construct, so oxlint silently loses coverage here. As a knock-on effect, an `eslint-disable no-multi-assign` directive that suppresses only the declarator-initializer form is reported as unused.

### oxlint version

1.78.0; originally observed on 1.69.0, re-verified on 1.72.0 and 1.78.0.

### Reproduction

`repro.ts`:

```ts
class C {
	hash: unknown
	done = false
	run() {
		const currentLoadHash = (this.hash = {}) // line 5 — should be flagged
		const loadComplete = (this.done = true) // line 6 — should be flagged
		let a, b, c
		a = b = c = 1 // line 8 — classic chain, IS flagged
		return [currentLoadHash, loadComplete, a, b, c]
	}
}
```

`.oxlintrc.json`:

```json
{
	"plugins": [],
	"categories": {"correctness": "off"},
	"rules": {"no-multi-assign": ["error"]}
}
```

Run:

```
oxlint -c .oxlintrc.json --disable-nested-config repro.ts
```

### Expected

All four assignments are flagged: the two variable-declarator forms at 5:28 and 6:25, plus the classic chain at 8:7 and 8:11. This matches ESLint, whose `no-multi-assign` walks the assignment regardless of whether it appears as a declarator initializer (including when wrapped in parentheses):

- ESLint 7.23.0 (with `@typescript-eslint/parser` 6.21) flags all four locations — verified.
- ESLint 9.39.4 (flat config, `typescript-eslint` 8.62.1 parser) flags the identical four locations — verified.

### Actual

oxlint 1.78.0 reports only the classic chain:

```
repro.ts:8:7: error eslint(no-multi-assign): Do not use chained assignment
repro.ts:8:11: error eslint(no-multi-assign): Do not use chained assignment
```

Lines 5 and 6 produce no diagnostic. On a file carrying `/* eslint-disable no-multi-assign */` purely to suppress the declarator-initializer forms, oxlint (with `--report-unused-disable-directives`) reports "Unused eslint-disable directive (no problems were reported)", confirming the rule never visits the construct.

Enabling the bundled real ESLint implementation as a jsPlugin (`eslint-js/no-multi-assign`) does flag both lines, so the parser/AST is capable; the gap is in the native rule's node matching.

### Cross-checked

| Engine | Flags `const x = (obj.y = z)` | Flags `a = b = c = 1` |
| --- | --- | --- |
| ESLint 7.23.0 (`@typescript-eslint/parser` 6.21) | yes (5:28, 6:25) | yes (8:7, 8:11) |
| ESLint 9.39.4 (`typescript-eslint` 8.62.1 parser) | yes (5:28, 6:25) | yes (8:7, 8:11) |
| oxlint 1.78.0 (native rule) | no | yes (8:7, 8:11) |
